src/plugins/experimental-discord/createGraph.js

```diff
@@ -236,43 +240,49 @@ function sharedReactionWeights({
   return hasEdges;
 }
 
-function asymptoticReactionWeights({
-  guild,
-  emojiWeights,
-  roleWeightConfig,
-  wg,
-  memberMap,
-  message,
-  reactions,
-  useAsymptoticReactionWeights,
-}) {
+/* This function prepares data fetched in the createGraph function for asymptotic
+dropoff in weights for multiple reactions from a single member.
+
+The reactions fetched in the createGraph function are sorted from highest to lowest,
+and separated into sub arrays of reactions grouped by reacting member.
+
+Each of these member specific reaction arrays are then run through the createReactionEdgesAndNodes
+function */
+function prepareAsymptoticReactionWeights(args): boolean {
+  const {emojiWeights, reactions} = args;
+  let {hasEdges} = args;
   // sort reactions from highest to lowest weight emojis
   const sortedReactions = [...reactions].sort((a, b) => {
-    const bEmojiWeight = emojiWeights[Model.emojiToRef(b.emoji)] || 1;
-    const aEmojiWeight = emojiWeights[Model.emojiToRef(a.emoji)] || 1;
+    const bEmojiWeight = NullUtil.orElse(
+      emojiWeights[Model.emojiToRef(b.emoji)],
+      1
+    );
+    const aEmojiWeight = NullUtil.orElse(
+      emojiWeights[Model.emojiToRef(a.emoji)],
+      1
+    );
     return bEmojiWeight - aEmojiWeight;
   });
 
   // separate reactions into groups per member (retains sorted order)
-  const mapMemberIdToReactions = new Map();
+  const memberIdToReactionsMap = new Map();
   sortedReactions.forEach((reaction) => {
-    const authorData = mapMemberIdToReactions.get(reaction.authorId) || [];
+    const authorData = NullUtil.orElse(
+      memberIdToReactionsMap.get(reaction.authorId),
+      []
+    );
     authorData.push(reaction);
-    mapMemberIdToReactions.set(reaction.authorId, authorData);
+    memberIdToReactionsMap.set(reaction.authorId, authorData);
   });
 
-  for (const memberReactions of mapMemberIdToReactions.values()) {
-    return sharedReactionWeights({
-      guild,
-      emojiWeights,
-      roleWeightConfig,
-      wg,
-      memberMap,
-      message,
+  for (const memberReactions of memberIdToReactionsMap.values()) {
+    hasEdges = createReactionEdgesAndNodes({
+      ...args,
+      hasEdges,
       reactions: memberReactions,
-      useAsymptoticReactionWeights,
     });
   }
+  return hasEdges;
```

Comment:
Rather than continually setting `hasEdges` as true as each iteration runs, let's revert to initializing it inside `createReactionEdgesAndNodes`, and in  `prepareAsymptoticReactionWeights` use `hasEdges` or better yet something more accurately named like `atleastOneNodeHasedges` as a flag that is only flipped `true` without passing it through both of these functions.